modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/ast/ApplicationExtensionModelsEnricher.java

```diff
@@ -0,0 +1,113 @@
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.internal.ast;
+
+import static org.mule.runtime.api.component.TypedComponentIdentifier.ComponentType;
+import static org.mule.runtime.api.component.TypedComponentIdentifier.ComponentType.OPERATION_DEF;
+import static org.mule.runtime.api.dsl.DslResolvingContext.getDefault;
+import static org.mule.runtime.core.api.util.boot.ExtensionLoaderUtils.getOptionalLoaderById;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_ARTIFACT_AST_PROPERTY_NAME;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_EXTENSION_NAME_PROPERTY_NAME;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_APPLICATION_LOADER_ID;
+import static org.mule.runtime.extension.api.ExtensionConstants.VERSION_PROPERTY_NAME;
+import static org.mule.runtime.extension.api.loader.ExtensionModelLoadingRequest.builder;
+
+import static java.util.Collections.singleton;
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.ast.api.ArtifactAst;
+import org.mule.runtime.ast.api.ArtifactType;
+import org.mule.runtime.extension.api.loader.ExtensionModelLoader;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+
+/**
+ * An {@link ArtifactExtensionModelsEnricher} suitable for the context of applications.
+ *
+ * @since 4.5.0
+ */
+public class ApplicationExtensionModelsEnricher implements ArtifactExtensionModelsEnricher {
+
+  private static final Logger LOGGER = getLogger(ArtifactAstUtils.class);
+
+  // TODO W-11796759: This class shouldn't know which are the specific reusable components.
+  private static final Set<ComponentType> REUSABLE_COMPONENT_TYPES = singleton(OPERATION_DEF);
+
+  private final String artifactId;
+  private final Optional<String> artifactVersion;
+
+  public ApplicationExtensionModelsEnricher(String artifactId, Optional<String> artifactVersion) {
+    this.artifactId = artifactId;
+    this.artifactVersion = artifactVersion;
+  }
+
+  @Override
+  public boolean applicable(ArtifactAst ast) {
+    return ast.topLevelComponentsStream()
+        .anyMatch(component -> REUSABLE_COMPONENT_TYPES.contains(component.getComponentType()));
+  }
+
+  @Override
+  public Set<ExtensionModel> getEnrichedExtensionModels(ArtifactAst ast, ClassLoader classLoader,
+                                                        Set<ExtensionModel> extensions) {
+    Optional<ExtensionModel> artifactExtensionModel =
+        parseApplicationExtensionModel(ast, classLoader, extensions);
+
+    if (artifactExtensionModel.isPresent()) {
```

Comment:
This piece of code is no longer present in the most recent version.
That being said, I similar fragment now appears [here](https://github.com/mulesoft/mule/pull/11776/files#diff-8a588bb1806dd53f441fcbda4dfe927efa22ad5e5deeabe5126e0016b41ab4e2R55), but in this case I think it will be a lot easier to read without using the functional style.
Let me know if you still would like me to change that new one.